bindings/guile/gnc-kvp-guile.cpp

```diff
@@ -59,16 +73,31 @@ gnc_scm_to_kvp_value_ptr(SCM val)
     {
         return new KvpValue{gnc_scm_to_utf8_string(val)};
     }
-    else if (SWIG_IsPointerOfType(val, SWIG_TypeQuery("_p_KvpFrame")))
+    else if (scm_is_list_of_string_pairs (val))
+    {
+        auto frame = new KvpFrame;
+        for (; !scm_is_null (val); val = scm_cdr (val))
+        {
+            auto key_str = scm_to_utf8_stringn (scm_caar (val), nullptr);
+            auto val_scm = scm_cdar (val);
+            auto prev = frame->set ({key_str}, gnc_scm_to_kvp_value_ptr (val_scm));
+            g_free (key_str);
+            // there is a pre-existing key-value
+            if (prev)
+                delete prev;
+        }
+        return new KvpValue (frame);
+    }
+    else if (scm_is_list (val))
```

Comment:
Hmm, if a NULL value is still a list then this would become the default result when `val` is NULL. Is that really desirable?